### PR TITLE
misc: address coverity false positives

### PR DESCRIPTION
**Pull request by amitkumar50**
Fixes the coverity issues:

** 1409838 Uninitialized scalar field
>CID 1409838 (#1-2 of 2): Uninitialized scalar field (UNINIT_CTOR)
>2. uninit_member: Non-static class member m_watchers_ret_val is
not initialized in this constructor nor in any functions that it calls.

** 1405850 Uninitialized scalar field
>2. uninit_member: Non-static class member reserv_heap_data is not
initialized in this constructor nor in any functions that it calls.
>4. uninit_member: Non-static class member lim_heap_data is not
initialized in this constructor nor in any functions that it calls.
>CID 1405850 (#1-3 of 3): Uninitialized scalar field (UNINIT_CTOR)
>6. uninit_member: Non-static class member ready_heap_data is not
initialized in this constructor nor in any functions that it calls.

** 1413774 Uninitialized scalar field
>CID 1413774 (#1-2 of 2): Uninitialized scalar field (UNINIT_CTOR)
>2. uninit_member: Non-static class member m_size is not initialized
in this constructor nor in any functions that it calls.

Signed-off-by: Amit Kumar <amitkuma@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
please put related commits into a single PR.

File: src/dmclock/src/dmclock_server.h [hunk 1]
```diff
@@ -296,9 +296,9 @@ namespace crimson {
 	// an idle client becoming unidle
 	double                prop_delta = 0.0;
 
-	c::IndIntruHeapData   reserv_heap_data;
-	c::IndIntruHeapData   lim_heap_data;
-	c::IndIntruHeapData   ready_heap_data;
+	c::IndIntruHeapData   reserv_heap_data{};
```
Comment (dillaman) [inline src/dmclock/src/dmclock_server.h]: Huh? If they can be default-initialized, why is this change needed?

Comment (tchaikov) [reply src/dmclock/src/dmclock_server.h]: strictly speaking, they are not "default-initialized". what is used here is "[zero-initialization](http://en.cppreference.com/w/cpp/language/zero_initialization)". we could use it without doing so explicitly, but it depends on how `ClientRec` is initialized. if we initialize it using "zero initialization", like

```c++
auto c = ClientRec{};
```
using the default ctor, then `{reserv,lim,ready}_heap_data`  are zero-initialized. see http://en.cppreference.com/w/cpp/language/value_initialization. but `ClientRec` has a user-defined ctor. so one way to initialize them is to use the "zero initialization" explicitly.

Comment (tchaikov) [reply src/dmclock/src/dmclock_server.h]: @amitkumar50 please read https://github.com/ceph/ceph/blob/master/src/dmclock/README.before-modifying-files-here-or-below and resubmit this change.

Comment (dillaman) [reply src/dmclock/src/dmclock_server.h]: Didn't realize it was just a typedef for ```size_t``` -- this is an excellent example of why this style of initialization is confusing.

Comment (tchaikov) [reply src/dmclock/src/dmclock_server.h]: i think the nice thing is that it's "confusing". because `IndIntruHeapData` is a typedef, i guess @ivancich wanted to hide the actual type of `IndIntruHeapData` from its users, so we can change it to other types (e.g, compound types) when necessary. in that case, the notation of `{}` will still work if they offer a default ctor.

Comment (ivancich) [reply src/dmclock/src/dmclock_server.h]: @tchaikov Thank you for that input. In my commit to the dmclock library to address this very same issue initialized with "= 0". But your point is well-taken and I'll look into changing it to "{}".

**Comment by dillaman** [comment]
@amitkumar50 ping

**Comment by dillaman** [comment]
Closing due to lack of activity
